src/test/rpc/ShardArchiveHandler_test.cpp

```diff
@@ -49,196 +49,207 @@ class ShardArchiveHandler_test : public beast::unit_test::suite
     {
-        testcase("testStateDatabase1");
+        testcase("testSingleDownloadAndStateDB");
 
-        {
-            beast::temp_dir tempDir;
-
-            auto c = jtx::envconfig();
-            auto& section = c->section(ConfigSection::shardDatabase());
-            section.set("path", tempDir.path());
-            section.set("max_size_gb", "100");
-            c->setupControl(true, true, true);
-
-            jtx::Env env(*this, std::move(c));
-            auto handler = RPC::ShardArchiveHandler::getInstance(
-                env.app(), env.app().getJobQueue());
-            BEAST_EXPECT(handler);
+        beast::temp_dir tempDir;
 
-            BEAST_EXPECT(handler->init());
+        auto c = jtx::envconfig();
+        auto& section = c->section(ConfigSection::shardDatabase());
+        section.set("path", tempDir.path());
+        section.set("max_size_gb", "100");
+        c->setupControl(true, true, true);
 
-            std::string const rawUrl = "https://foo:443/1.tar.lz4";
-            parsedURL url;
+        jtx::Env env(*this, std::move(c));
+        auto handler = env.app().getShardArchiveHandler();
+        BEAST_EXPECT(handler);
+        BEAST_EXPECT(dynamic_cast<RPC::RecoveryHandler*>(handler) == nullptr);
 
-            parseUrl(url, rawUrl);
-            handler->add(1, {url, rawUrl});
+        std::string const rawUrl = "https://foo:443/1.tar.lz4";
+        parsedURL url;
 
-            {
-                std::lock_guard<std::mutex> lock(handler->m_);
+        parseUrl(url, rawUrl);
+        handler->add(1, {url, rawUrl});
 
-                auto& session{handler->sqliteDB_->getSession()};
+        {
+            std::lock_guard<std::mutex> lock(handler->m_);
 
-                soci::rowset<soci::row> rs =
-                    (session.prepare << "SELECT * FROM State;");
+            auto& session{handler->sqliteDB_->getSession()};
 
-                uint64_t rowCount = 0;
+            soci::rowset<soci::row> rs =
+                (session.prepare << "SELECT * FROM State;");
 
-                for (auto it = rs.begin(); it != rs.end(); ++it, ++rowCount)
-                {
-                    BEAST_EXPECT(it->get<int>(0) == 1);
-                    BEAST_EXPECT(it->get<std::string>(1) == rawUrl);
-                }
+            std::uint64_t rowCount = 0;
 
-                BEAST_EXPECT(rowCount == 1);
+            for (auto it = rs.begin(); it != rs.end(); ++it, ++rowCount)
+            {
+                BEAST_EXPECT(it->get<int>(0) == 1);
+                BEAST_EXPECT(it->get<std::string>(1) == rawUrl);
             }
 
-            handler->release();
+            BEAST_EXPECT(rowCount == 1);
         }
 
-        // Destroy the singleton so we start fresh in
-        // the next testcase.
-        RPC::ShardArchiveHandler::instance_.reset();
+        handler->release();
     }
 
+    // Test the shard downloading module by initiating
+    // and completing three downloads and verifying and
```

Comment:
Extra "and"? "... and verifying ~and~ the contents ..."

Reply:
Done